app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,27 @@
+class SessionsController < ApplicationController
+  def new
+
+  end
+
+  def create
+
+    user = User.log_in(params[:email], params[:password])
+
+    if user
+      session[:user_id] = user.id
+
+      session[:order_id] = ((Order.where(user_id: user.id).length == 0) ? Order.create(user_id: session[:user_id], status: "pending").id : Order.where(user_id: user.id).last.id)
+
```

Comment:
I really like that you are retrieving the user's most recent order when they log back in. What if the order has already been purchased?